src/subscribe.rs

```diff
@@ -0,0 +1,429 @@
+#[ft_sdk::form]
+fn subscribe(
+    ft_sdk::Query(name): ft_sdk::Query<"name", Option<String>>,
+    ft_sdk::Query(phone): ft_sdk::Query<"phone", Option<String>>,
+    ft_sdk::Query(email): ft_sdk::Query<"email", Option<String>>,
+    ft_sdk::Query(source): ft_sdk::Query<"source", Option<String>>,
+    ft_sdk::Query(topic): ft_sdk::Query<"topic", Option<String>>,
+    ft_sdk::Query(next): ft_sdk::Query<"next", Option<String>>,
+    sid: ft_sdk::Cookie<{ ft_sdk::auth::SESSION_KEY }>,
+    host: ft_sdk::Host,
+    mountpoint: ft_sdk::Mountpoint,
+    mut conn: ft_sdk::Connection,
+) -> ft_sdk::form::Result {
+    use diesel::prelude::*;
+
+    let (subscriber, id) = validate(&mut conn, name, phone, email, sid)?;
+    let user_id = get_or_create_user_id(&mut conn, id, &subscriber)?;
```

Comment:
Merge these two and do something similar to `unsubscribe::get_user`. It's much cleaner.